### ISSUE TITLE
[BUG] Multiple inline math elements in different axes causing axis labels to disappear

### ISSUE THREAD

**Issue by golmschenk** [bokeh/bokeh#13607]
### Software versions

Python version        :  3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:37:07) [Clang 15.0.7 ]
IPython version       :  (not installed)
Tornado version       :  6.3.2
Bokeh version         :  3.3.2
BokehJS static path   :  /opt/homebrew/Caskroom/mambaforge/base/envs/short_period_variable_neural_network_paper/lib/python3.11/site-packages/bokeh/server/static
node.js version       :  v20.5.0
npm version           :  9.8.0
jupyter_bokeh version :  (not installed)
Operating system      :  macOS-14.2-arm64-arm-64bit

### Expected behavior

Multiple inline math elements should be rendered regardless of what the opposing axis in the figure contains.

### Observed behavior

In some configurations, when multiple inline math elements are included in both axes of a figure, both axis labels are not rendered.

### Example code

In this example, neither axis label is rendered.

```Python
from bokeh.io import show
from bokeh.plotting import figure as Figure

figure0 = Figure(x_axis_label=r'$$a$$\[b\]', y_axis_label=r'$$c$$\[d\]', width=200, height=200)
figure0.line(x=[0, 1], y=[0, 1])
show(figure0)
```
<img width="206" alt="image0" src="https://github.com/bokeh/bokeh/assets/1691031/a2bc0873-487f-4df7-af0a-17240421b967">

If the second inline math element is removed from either axis label, both axis labels are correctly rendered. That is if either `\[b\]` or `\[d\]` is removed, both labels are correctly rendered.

Notably, if the above figure with incorrect rendering is placed in a `Row` with a duplicate figure that had either `\[b\]` or `\[d\]` removed, then all 4 axis labels are correctly rendered. For example, the following code has a `Row` containing the failing figure above, but when placed in this `Row`, the figure correctly renders.

```Python
from bokeh.io import show
from bokeh.models import Row
from bokeh.plotting import figure as Figure

figure1 = Figure(x_axis_label=r'$$a$$\[b\]', y_axis_label=r'$$c$$', width=200, height=200)
figure1.line(x=[0, 1], y=[0, 1])
figure0 = Figure(x_axis_label=r'$$a$$\[b\]', y_axis_label=r'$$c$$\[d\]', width=200, height=200)
figure0.line(x=[0, 1], y=[0, 1])
row = Row(figure1, figure0)
show(row)
```
<img width="403" alt="image1" src="https://github.com/bokeh/bokeh/assets/1691031/c3456cdf-958d-49f5-96ce-1d418bce9140">

**Comment by mattpap** [comment]
For reference, this is how the first plot looks like in 3.4-dev.4:
![image](https://github.com/bokeh/bokeh/assets/27475/c9336405-023a-4917-ae99-9e6f8146e4dc)

**Comment by mosc9575** [comment]
@mattpap This is odd. I also tried to reproduce and it looks like it it was working with `3.4.0-dev.3`

![grafik](https://github.com/bokeh/bokeh/assets/68053396/d1829385-93d3-4845-aa58-801e5f80c7b2)

Update:

After an update to `3.4.0-dev.4` I get the same result as above. Can't reproduce.

![grafik](https://github.com/bokeh/bokeh/assets/68053396/68a6271a-80b8-4069-b1d2-bcc30b866f39)

**Comment by golmschenk** [comment]
@mosc9575, are you unable to reproduce it when not using Jupyter? I rarely use Jupyter, but I tried running this in a Colab session (using `3.4.0-dev.4`) and found that I was getting inconsistent results. Occasionally the result would appear as @mattpap showed, but then when I refreshed the site, or re-ran the cell, the labels were correctly rendered. I'm guessing there must be some stateful or race condition.

https://colab.research.google.com/drive/1MlYGQFcdhCJjIC3bTdur-GkdTx9Y51xF?usp=sharing

<img width="850" alt="Screenshot 2023-12-16 at 17 14 17" src="https://github.com/bokeh/bokeh/assets/1691031/7625c0bc-d8da-4066-9710-2b317cbd6371">
